controller/controller_test.go

```diff
@@ -912,20 +940,23 @@ func TestControllerMutation(t *testing.T) {
 				},
 				Status: statusAssigned([]string{"1.2.3.0"}),
 			},
-			wantErr: true,
+			expectedSyncState: controllers.SyncStateSuccess,
 		},
 	}
 
 	for i := 0; i < 100; i++ {
 		for _, test := range tests {
 			t.Run(test.desc, func(t *testing.T) {
 				k.reset()
+				c.ips.Unassign("test")
```

Comment:
By unassigning here, we are making the tests stateless. The outer loop is there so the execution of the various SetBalancer is tested with shuffled order. 

Is there a reason for clearing the state and enforcing the previous allocation for some tests?